money-core/src/main/scala/com/comcast/money/core/internal/SpanLocal.scala

```diff
@@ -34,42 +33,31 @@ trait SpanContext {
  */
 object SpanLocal extends SpanContext {
 
-  // A stack of span ids for the current thread
-  private[this] val threadLocalCtx = new ThreadLocal[List[Span]]
-
   private lazy val mdcSupport = new MDCSupport()
 
   import mdcSupport._
 
-  override def current: Option[Span] = Option(threadLocalCtx.get).flatMap(_.headOption)
+  override def current: Option[Span] = fromContext(Context.current)
 
-  override def push(span: Span): Unit =
+  override def push(span: Span): Scope =
```

Comment:
The removal of `pop` would definitely be a breaking change for anyone who directly took a dependency on `SpanLocal` or `SpanContext`, which are internal APIs.

Reply:
Can we make those package private as part of this PR?

Reply:
Making these package private might be tricky as a number of the other projects use `SpanLocal` directly and `SpanContext` is exposed directly from `Tracer`.  Although, with the exception of `clear`, all of the members should be accessible through the `Tracer` object instead.  I can look into it but it might be more involved.

Reply:
Okay. I wasn't sure if we could make it private to money and still have access in all the other packages